jsonrpc/ipc_test.go

```diff
@@ -0,0 +1,260 @@
+package jsonrpc_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func unixDial(ctx context.Context, endpoint string) (net.Conn, error) {
+	return new(net.Dialer).DialContext(ctx, "unix", endpoint)
+}
```

Comment:
```suggestion
func unixDial(t *testing.T, ctx context.Context, endpoint string) net.Conn {
	c, err := new(net.Dialer).DialContext(ctx, "unix", endpoint)
	require.NoError(t, err)
	return c
}
```
This may simplify some of the tests.

edit: I see there are asserts for specific errors. Maybe those cases could forgo the helper and call `DialContext` directly?